packages/core/src/authorization.ts

```diff
@@ -0,0 +1,28 @@
+import {Ctx} from "./middleware"
+import {AuthenticatedSessionContext} from "./supertokens"
+import {ZodSchema, infer as zInfer} from "zod"
+
+export type ProtectArgs<T> = {schema?: T; authorize?: boolean | unknown}
+
+interface AuthenticatedCtx extends Ctx {
+  session: AuthenticatedSessionContext
+}
+
+export const protect = <T extends ZodSchema<any, any>, U = zInfer<T>>(
+  {schema, authorize = true}: ProtectArgs<T>,
+  resolver: (args: U, ctx: AuthenticatedCtx) => any,
+) => {
+  return (rawInput: U, ctx: Ctx) => {
+    if (authorize) {
+      const authorizeInput: any[] = ["superadmin", "SUPERADMIN"]
+      if (Array.isArray(authorize)) {
+        authorizeInput.push(...authorize)
+      } else if (typeof authorize !== "boolean") {
+        authorizeInput.push(authorize)
+      }
+      ;(ctx as any).session.authorize(authorizeInput)
```

Comment:
This doesn't leave anyway to call session.authorize() with no params (ensure logged in user without caring about what role they are).

Reply:
I guess it sounds like you're solving for usage to be that you need to explicitly list all roles you want to allow instead of the opposite (what I'm suggesting) where the default is empty authorize() call and only add roles if you need it.

Not clear to me what the better developer experience is but I guess your direction is safer in the sense that everything is locked down by default.  It does mean if I add a new role I then have to go back and add it everywhere I use protect.  That should happen pretty rarely though so not a big deal.  I guess you can also just have a roles util where you export constants like ALL_ROLES, ALL_ADMIN_ROLES, MANAGEMENT_ROLES, etc and just protect using those and then can just update the actual roles in one place.

Reply:
Yeah, and this again points to why we really need a more permissions-based model: https://github.com/blitz-js/blitz/issues/756